cleanup/mavenSnapshotCleanupWhenRelease/mavenSnapshotCleanupWhenRelease.groovy

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2014 JFrog Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import org.artifactory.fs.ItemInfo
+import org.artifactory.repo.RepoPathFactory
+
+import groovy.transform.Field
+
+@Field final String PROPERTIES_FILE_PATH = "plugins/${this.class.name}.properties"
+
+def config = new ConfigSlurper().parse(new File(ctx.artifactoryHome.haAwareEtcDir, PROPERTIES_FILE_PATH).toURI().toURL())
+def repositoriesMSCWR = [:]
+
+log.info "Check 'Maven Snapshot Cleanup When Release' plugin repositories list: $config.repositories"
+
+config.repositories.each{ repositorySettings ->
+
+    def release = repositorySettings[ 0 ] ? repositorySettings[ 0 ] as String : ["undefined"]
+    def snapshot = repositorySettings[ 1 ] ? repositorySettings[ 1 ] as String : ["undefined"]
+
+    def releaseConfig = repositories.getRepositoryConfiguration(release)
+    def sapshotConfig = repositories.getRepositoryConfiguration(snapshot)
+
+    if (releaseConfig && 'maven' == releaseConfig.getPackageType() && releaseConfig.isHandleReleases() && sapshotConfig && 'maven' == sapshotConfig.getPackageType() && sapshotConfig.isHandleSnapshots()){
+        repositoriesMSCWR[ release ] = snapshot
+    }else{
+        log.error "Skip repositories couple $release/$snapshot (does not exist or incorrect handle policy / package type)"
+    }
+}
+
+storage {
+    afterCreate { ItemInfo item ->
```

Comment:
@elioengcomp: The snapshot->release promotion (copy/move) is not really sustainable IMO. You have always some jobs todo for creating a release (at min remove the -SNAPSHOT suffix in pom content ... rarely only on the artifact version).

But I can support these use cases (need test the difference between `afterCreate`, `afterMove`, `afterCopy`).